src/raft.rs

```diff
@@ -223,8 +223,8 @@ pub fn quorum(total: usize) -> usize {
 
 impl<T: Storage> Raft<T> {
     /// Creates a new raft for use on the node.
-    pub fn new(c: &Config, store: T) -> Raft<T> {
-        c.validate().expect("configuration is invalid");
+    pub fn new(c: &Config, store: T) -> Result<Raft<T>> {
+        c.validate()?;
         let rs = store.initial_state().expect("");
```

Comment:
Maybe we can also push up this expect?

Reply:
My intention here was to propagate the `Error:ConfigInvalid` returned by `validate()`. https://github.com/pingcap/raft-rs/blob/13ec5ac33a3800f89a6946cb8c54ceba4b84ab98/src/config.rs#L170

I can make the test case more explicit to show that? Changing my test case to the below would show that it propagates back up to the user.
```
fn test_new_raft_with_bad_config_errors() {
    let invalid_config = new_test_config(INVALID_ID, vec![1, 2], 1, 1);
    let raft = Raft::new(&invalid_config, new_storage());
    assert_eq!(
        raft.err(),
        Some(Error::ConfigInvalid("invalid node id".to_owned()))
    )
}
```

Reply:
Oh, yes, that's fine. I meant the `store.initial_state()` could probably use a `?` as well instead of an `expect()`.

Reply:
Oh! Yes, sorry about that; I'll change that as well.